src/mono/mono/mini/mini-llvm.c

```diff
@@ -8433,7 +8433,21 @@ MONO_RESTORE_WARNING
 			LLVMValueRef dest;
 
 			dest = convert (ctx, LLVMBuildAdd (builder, convert (ctx, values [ins->inst_destbasereg], IntPtrType ()), LLVMConstInt (IntPtrType (), ins->inst_offset, FALSE), ""), pointer_type (t));
-			mono_llvm_build_aligned_store (builder, lhs, dest, FALSE, 1);
+			if (mono_class_value_size (ins->klass, NULL) == 12) {
```

Comment:
Why do we use expect that if we encounter a value type with size 12, it must be a Vector3? 
Couldn't there be a value type with size 12 that is just a plain struct, thus shouldn't be handled as a vector.

This comment is not directly related to this PR as we are using the `mono_class_value_size (ins->klass, NULL) == 12` elsewhere as well for the same check, so it is probably correct. I'm just trying to understand the logic behind.

Reply:
I think we only use this check in op codes dealing with SIMD vectors. When there is an ambiguity there is also an `mini_class_is_simd` check before. I agree this can be made better, but I would do this in separate PR.

Reply:
Ah right, that makes sense. I wouldn't even propose to change it, I was just getting curious :)